### PR TITLE
Allow to not redirect to a language

### PR DESCRIPTION
**Pull request by aschempp**
Using this new flag, one can configure a root page to not redirect to the language. Be aware that this is **not** the same as `doNotRedirectEmpty`, quite the contrary. 

It is now possible to define a root page with empty URL prefix, and several others with URL prefix. The user will always see the "empty prefix" page without being redirected when enabling the checkbox on all root pages.

### CODE REVIEW

**Comment by fritzmg** [comment]
👍 

So you have to enable it for all website roots of the same domain, correct?

**Comment by aschempp** [comment]
> So you have to enable it for all website roots of the same domain, correct?

You don't _have_ to, e.g. if you want everyone to be redirected to `/en` even if `/de` exists. But not sure if that would make sense 😆

File: core-bundle/src/Resources/contao/models/PageModel.php [hunk 1]
```diff
@@ -28,6 +28,7 @@
  * @property string  $type
  * @property string  $pageTitle
  * @property string  $language
+ * @property boolean $doNotRedirectLanguage
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/models/PageModel.php]: Please also add the `find*ByDoNotRedirectLanguage()` and `countByDoNotRedirectLanguage()` methods.

**Comment by leofeyer** [comment]
| Page | URL prefix | language |
| --- | --- | --- |
| Page A | - | ??? |
| Page B | en | English |
| Page C | de | German |

So the new option allows me to browse Page A (e.g. under `https://domain.com`) without being redirected to Page B or C, even if `en` or `de` were my preferred browser language. Correct?

File: core-bundle/src/Resources/contao/models/PageModel.php (hunk 1 above)
Comment (aschempp) [reply core-bundle/src/Resources/contao/models/PageModel.php]: I think they're pretty stupid and unnecessary bloat, but added in c50e0ff49a880e92b3cff8caf30643ef1f572db5 🙃

**Comment by aschempp** [comment]
> So the new option allows me to browse Page A (e.g. under `https://domain.com`) without being redirected to Page B or C, even if `en` or `de` were my preferred browser language. Correct?

That's correct. The language of Page A can be english as well, or just the language you want to be used (by the language files).

**Comment by aschempp** [comment]
FYI I'm not sure the palette "looks good" this way?

Bildschirmfoto 2020-07-29 um 10 53 33

**Comment by leofeyer** [comment]
> > So the new option allows me to browse Page A (e.g. under `https://domain.com`) without being redirected to Page B or C, even if `en` or `de` were my preferred browser language. Correct?
> 
> That's correct. The language of Page A can be english as well, or just the language you want to be used (by the language files).

Can you elaborate on the use-case? Why would I want to browse `https://domain.com` if there is `https://domain.com/de/`?

**Review by leofeyer — CHANGES_REQUESTED** [review]
I do not like the fact that in order to configure the behavior of Page A (namely not to redirect me), I have to edit Page B and Page C. That does not seem very intuitive, does it? Maybe we can come up with a better solution.

**Comment by fritzmg** [comment]
Hm, yeah, shouldn't it work automatically anyway? If you have the following website roots:

* A: `https://example.org`
* B: `https://example.org/en`
* C: `https://example.org/de`

and a request to `https://example.org/` is made, shouldn't match website root `A` anyway? Regardless of the `Accept-Language` header of the request.

**Comment by fritzmg** [comment]
> Can you elaborate on the use-case? Why would I want to browse `https://domain.com` if there is `https://domain.com/de/`?

e.g.

* https://www.apple.com/ (`lang="en-US"`)
* https://www.apple.com/at/ (`lang="de-AT"`)
* https://www.apple.com/de/ (`lang="de-DE"`)
* https://www.apple.com/uk/ (`lang="en-GB"`)
* …

**Comment by leofeyer** [comment]
This would be the "do not redirect empty" case, however, the initial issue description says:

> Be aware that this is **not** the same as `doNotRedirectEmpty`, quite the contrary.

🤷🏻‍♂️

**Comment by aschempp** [comment]
> This would be the "do not redirect empty" case, however, the initial issue description says:
> 
> > Be aware that this is **not** the same as `doNotRedirectEmpty`, quite the contrary.
> 
> 🤷🏻‍♂️

No that's not true! `doNotRedirectEmpty` would **serve the UK page at /** instead of redirecting. This PR **always** serves the US page on the root domain.

**Comment by leofeyer** [comment]
Ok, so this is in fact the apple.com case?

**Comment by aschempp** [comment]
> Ok, so this is in fact the apple.com case?

yes, any page that does **not** automatically redirect you to the preferred language.

Thinking about it, maybe we should invert the checkbox and enable it by default. Like _Automatically redirect a visitor to this language if it's the preferred one_?

**Comment by leofeyer** [comment]
> Thinking about it, maybe we should invert the checkbox and enable it by default. Like _Automatically redirect a visitor to this language if it's the preferred one_?

That does not make it less cumbersome. I still only want to configure Page A and I still can only do it by configuring Page B, C, D and so on. Is there no way to set the flag on Page A and adjust the router accordingly?

**Comment by aschempp** [comment]
Why would you want to configure Page A? You want to tell Page B that it should never match a language. That's not related to Page A. You can redirect two languages but not a third.

**Comment by leofeyer** [comment]
This is how I would arrange the new field:



@aschempp Any objections?

**Review by leofeyer — APPROVED** [review]

**Pull request merged**